### PR TITLE
Set up Nix Flake for kup tool

### PR DESCRIPTION
**Pull request by ehildenb**
~Blocked on: https://github.com/runtimeverification/plutus-core-semantics/pull/357~
~Blocked on: https://github.com/runtimeverification/plutus-core-semantics/pull/358~

- [x] Get Nix flake figured out.
- [ ] Update status check "Bulid and Test KPlutus => Build and Test KPlutus".
- [ ] Add Nix flake status checks.

### CODE REVIEW

**Comment by ehildenb** [comment]
@Baltoli if you have a moment, can you take a look at this one? It's failing with the "hatchling" nix error.

**Comment by Baltoli** [comment]
Hmm, I haven't seen this one before. I'll dig into it in some more detail if @goodlyrottenapple doesn't recognise it off the top of his head.

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+  inputs = {
+    k-framework.url = "github:runtimeverification/k";
+    nixpkgs.follows = "k-framework/nixpkgs";
+    flake-utils.follows = "k-framework/flake-utils";
+    rv-utils.url = "github:runtimeverification/rv-nix-tools";
+    poetry2nix.follows = "pyk/poetry2nix";
+    blockchain-k-plugin.url =
+      "github:runtimeverification/blockchain-k-plugin/3010915ef37b0e614528821a7c8717c797a365a6";
+    blockchain-k-plugin.inputs.flake-utils.follows = "k-framework/flake-utils";
```
Comment (ehildenb) [inline flake.nix:12]:
```suggestion
    blockchain-k-plugin.url = "github:runtimeverification/blockchain-k-plugin/3010915ef37b0e614528821a7c8717c797a365a6";
    blockchain-k-plugin.inputs.flake-utils.follows = "k-framework/flake-utils";
```

Comment (ehildenb) [reply flake.nix:12]: @goodlyrottenapple this has to be one-line or the sed expression in `.github/workflows/update-version.yml` needs to be updated.

**Review by goodlyrottenapple — APPROVED** [review]

**Pull request merged**